workflow/controller/operator.go

```diff
@@ -2174,7 +2174,7 @@ func getStepOrDAGTaskName(nodeName string) string {
 	if strings.Contains(nodeName, ".") {
 		name := nodeName[strings.LastIndex(nodeName, ".")+1:]
 		// Retry, withItems and withParam scenario
-		if indx := strings.LastIndex(name, "("); indx > 0 {
+		if indx := strings.Index(name, "("); indx > 0 {
```

Comment:
Recent Retry logic refactoring added one more `(` in node name '.divide-by-2(2:3)(0)'. LastIndex won't for that. We should use an Index which will handle retry and non-retry scenario

Reply:
test?

Reply:
@alexec added test